File: 프로그래머스/lv3/43162. 네트워크/네트워크.py
from collections import deque
def solution(n, computers):
    
    def BFS(i):
        queue=deque()
        queue.append(i)
        
        while queue:
            now = queue.popleft()
            #방문 체크
            visit[now] = 1
            
            for i in range(n):
                if not visit[i] and computers[now][i]:
                    queue.append(i)
        print(visit)
        
    answer = 0
    # visit 배열을 잘못 만들었었음.. 바보..
    visit = [0]*n
    for i in range(n):
        #방문한 적이 없으면
        if not visit[i]:
            BFS(i)
            answer+=1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function  BFS(n, start,visited, computers, count){
    const queue = [start]
    visited[start] = count
    
    while(queue.length>0){
        const cur = queue.shift()
        
        for(let i=0;i<n;i++){
            if(computers[cur][i] === 1 && visited[i] === 0){
                queue.push(i)
                visited[i] = count
            }
        }
    } 
}

function solution(n, computers) {
    var answer = 0;
    const visited = Array(n).fill(0)  
    
    for(let i=0;i<n;i++){
        if(visited[i] === 0){
            answer+=1;
            BFS(n, i, visited, computers, answer)
        }
    }
    return answer;
}